lib/bitrange.h

```diff
@@ -51,4 +58,686 @@ class bitranges {
     iter end() const { return iter(bits.end()); }
 };
 
+class JSONGenerator;
+class JSONLoader;
+
+/**
+ * @return the result of dividing @dividend by @divisor, rounded towards
+ * negative infinity. This is different than normal C++ integer division, which
+ * rounds towards zero. For example, `-7 / 8 == 0`, but
+ * `divideFloor(-7, 8) == -1`.
+ */
+inline int divideFloor(int dividend, int divisor) {
```

Comment:
Can these be moved to some separate header? E.g. `math_extras.h` or so? And / or put into some `detail` namespace instead?

Reply:
I've kept them in bitrange.h for now but put them in a BitRange::Detail namespace.